ReleaseProcessAutomation/Jira/CredentialManagement/JiraCredentialManager.cs

```diff
@@ -0,0 +1,143 @@
+using ReleaseProcessAutomation.Jira.Authentication;
+using ReleaseProcessAutomation.Jira.Utility;
+using ReleaseProcessAutomation.ReadInput;
+using Serilog;
+using Spectre.Console;
+
+namespace ReleaseProcessAutomation.Jira.CredentialManagement;
+
+public class JiraCredentialManager
+    : IJiraCredentialManager
+{
+  private readonly Config _config;
+  private readonly IInputReader _inputReader;
+  private readonly IAnsiConsole _console;
+  private readonly IJiraAuthenticator _jiraAuthenticator;
+  private readonly ILogger _log = Log.ForContext<JiraCredentialManager>();
+
+  public JiraCredentialManager (
+      Config config,
+      IInputReader inputReader,
+      IAnsiConsole console,
+      IJiraAuthenticator jiraAuthenticator)
+  {
+    _config = config;
+    _inputReader = inputReader;
+    _console = console;
+    _jiraAuthenticator = jiraAuthenticator;
+  }
+
+  public Credentials GetCredential (string target)
+  {
+    var cred = CredentialManager.GetCredentials(target);
+
+    if (cred == null)
+      return AskForCredentials(target);
+
+    var credentials = new Credentials
+                      {
+                          Username = cred.UserName,
+                          Password = cred.Password
+                      };
+
+    try
+    {
+      CheckJiraCredentials(credentials);
+    }
+    catch
+    {
+      _console.WriteLine("Invalid Jira Credentials saved.");
+      return AskForCredentials(target);
+    }
+
+    return credentials;
+  }
+
+  private Credentials AskForCredentials (string target)
+  {
+    while (true)
+    {
+      var tmpCredentials = new Credentials
+                           {
+                               Username = _inputReader.ReadString("Please enter your Jira username"),
+                               Password = _inputReader.ReadHiddenString("Please enter your Jira password")
+                           };
+
+      try
+      {
+        CheckJiraCredentials(tmpCredentials);
+      }
+      catch (Exception e)
+      {
+        _console.WriteLine("The input credentials didnt match, do you want to try again?");
+        if (_inputReader.ReadConfirmation())
+          continue;
+        throw new JiraAuthenticationException("Authentication not successful, user does not want to try again.", e);
+      }
+
+      _console.WriteLine("Do you want to save the login information to the credential manager?");
+      if (_inputReader.ReadConfirmation())
+      {
+        SaveCredentials(tmpCredentials, target);
+
+        const string message = "Saved Password";
+        _console.WriteLine(message);
+        _log.Information(message);
+      }
+
+      return tmpCredentials;
+    }
+  }
+
+  private void SaveCredentials (Credentials tmpCredentials, string target)
+  {
+    var cred = new NetworkCredential(tmpCredentials.Username, tmpCredentials.Password);
+    try
+    {
+      CredentialManager.RemoveCredentials(target);
+    }
+    catch
+    {
+      // ignored
```

Comment:
We should document why this is not a problem and why we ignore it. As it stands the comment does not add any value